### PR TITLE
Fixed potential crash in hs.serial when reloading Hammerspoon

### LINKED ISSUE #2763: Crash in hs.serial
This might be a bug in the underlying [ORSSerialPort](https://github.com/armadsen/ORSSerialPort) pod?

CC @armadsen in case he's interested.

Crash log: https://sentry.io/share/issue/57c1cd8f2df7404bb31ff5141a5b902a/

### PR DESCRIPTION
**Pull request by latenitefilms**
- Closes #2763

### CODE REVIEW

File: scripts/docs/bin/build_docs.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python3
+#!/usr/bin/env python3
```
Comment (cmsj) [inline scripts/docs/bin/build_docs.py:1]: I know it makes everyone but my life harder to use the OS provided python3, but I want it to stay that way - I don't want to have to try and make sure that the same python environment is being used by makefiles, pip installs, local builds, CI builds, release scripts, etc.

File: Hammerspoon Tests/HSserial.m [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- (void)setUp {
+    [super setUpWithRequire:@"test_serial"];
+}
+
+- (void)tearDown {
+    [super tearDown];
+}
+
+- (void)testAvailablePortNames {
+    RUN_LUA_TEST()
```
Comment (cmsj) [inline Hammerspoon Tests/HSserial.m:22]: I think these are all going to need a SKIP_IN_GITHUB_ACTIONS() before the RUN_LUA_TEST()s. It's awesome that you've added tests, but it looks like they depend on /dev/cu.MALS existing, which I'm guessing is the hardware you're using this with.

I've never been able to figure out what it is, but my Mac Pro seems to have a /dev/cu.pci-serial2, so maybe we can add the skips and then in the Lua test file, figure out which of the two serial devices to use - at least that way the test will pass on both your machine and mind :)

**Review by cmsj — CHANGES_REQUESTED** [review]

File: scripts/docs/bin/build_docs.py (hunk 1 above)
Comment (latenitefilms) [reply scripts/docs/bin/build_docs.py:1]: Ah, yes, you can revert this when merging. I'm still on Mojave, so I needed to make the change to make Hammerspoon build.

File: Hammerspoon Tests/HSserial.m (hunk 2 above)
Comment (latenitefilms) [reply Hammerspoon Tests/HSserial.m:22]: `/dev/cu.MALS` should be built into macOS - do you not see it on your machine?

Comment (latenitefilms) [reply Hammerspoon Tests/HSserial.m:22]: I've changed to `Bluetooth-Incoming-Port`, which I'm HOPING is on most/all Mac's?

Comment (cmsj) [reply Hammerspoon Tests/HSserial.m:22]: yeah I have the bluetooth incoming one, but no MALS. Weird

Comment (asmagill) [reply Hammerspoon Tests/HSserial.m:22]: FWIW, I don't have the MALS one, either... what device does it correspond to?

Comment (latenitefilms) [reply Hammerspoon Tests/HSserial.m:22]: Maybe it's something to do with MacBook Pro's - connection to the Touch Bar?

Comment (asmagill) [reply Hammerspoon Tests/HSserial.m:22]: If so, then it's an intel thing as my M1 doesn't haven't it either (and it does have the touchbar). Oh well, I do think that the bluetooth one is probably more reliably available, so my curiosity about the device isn't that important 😜

**Pull request merged**